include/aws/auth/credentials.h

```diff
@@ -797,6 +797,28 @@ struct aws_credentials *aws_credentials_new_from_string(
     const struct aws_string *session_token,
     uint64_t expiration_timepoint_seconds);
 
+/**
+ * Creates a new set of AWS credentials with account_id
+ *
+ * @param allocator memory allocator to use
+ * @param access_key_id  value for the aws access key id field
+ * @param secret_access_key value for the secret access key field
+ * @param session_token (optional) security token associated with the credentials
+ * @param account_id (optional) account id associated with the credentials
+ * @param expiration_timepoint_seconds timepoint, in seconds since epoch, that the credentials will no longer
+ * be valid past.  For credentials that do not expire, use UINT64_MAX
+ *
+ * @return a valid credentials object, or NULL
+ */
+AWS_AUTH_API
+struct aws_credentials *aws_credentials_new_from_string_with_account_id(
+    struct aws_allocator *allocator,
+    const struct aws_string *access_key_id,
+    const struct aws_string *secret_access_key,
+    const struct aws_string *session_token,
+    const struct aws_string *account_id,
+    uint64_t expiration_timepoint_seconds);
```

Comment:
There's a time-honored tradition in this codebase, when new arguments keep creeping into constructors...
```suggestion
struct aws_credentials *aws_credentials_new_from_options(
    struct aws_allocator *allocator,
    const struct aws_credentials_options *options);
```

Reply:
There are multiple types of credentials and they have new functions like `new_ecs`, `new_anonymous`, `new`, and `new_from_string`. Internally, they're represented with a union https://github.com/awslabs/aws-c-auth/blob/main/source/credentials.c#L80 and the actual credentials type is supposed to be opaque. I think consolidating everything into an option struct would be a mess. 

We can consolidate `new` and `new_with_account_id` function into an option struct and `new_from_string` and `new_from_string_with_account_id` into an option struct but we will potentially need new structs for new types.

Reply:
Thanks, updated with the new_from_options function.

Reply:
sorry I'm annoying I know